### PR TITLE
sys/hashes: add PSA Crypto wrappers

### PR DESCRIPTION
**Pull request by Einhornhool**
### Contribution description
This adds wrappers for the PSA Crypto module to RIOT hashes

### Issues/PRs references
Prerequisite: https://github.com/RIOT-OS/RIOT/pull/18578

Split up PR https://github.com/RIOT-OS/RIOT/pull/18547

### CODE REVIEW

File: sys/hashes/psa_riot_hashes/md5.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+psa_status_t psa_hashes_md5_finish(psa_hashes_md5_ctx_t *ctx,
+                                   uint8_t *hash,
+                                   size_t hash_size,
+                                   size_t *hash_length)
+{
+    md5_final((md5_ctx_t *)ctx, hash);
+
+    (void)hash_size;
```
Comment (leandrolanzieri) [inline sys/hashes/psa_riot_hashes/md5.c:44]: Probably also check that `MD5_DIGEST_LENGTH <= hash_size`

File: sys/hashes/psa_riot_hashes/md5.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @}
+ */
+
+#include "psa/crypto.h"
+#include "hashes/psa/riot_hashes.h"
+
+psa_status_t psa_hashes_md5_setup(psa_hashes_md5_ctx_t *ctx)
+{
+    md5_init((md5_ctx_t *)ctx);
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply sys/hashes/psa_riot_hashes/md5.c:25]: Also for the rest of the hashes

File: sys/hashes/psa_riot_hashes/hmac_sha256.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                                         uint8_t *mac,
+                                         size_t mac_size,
+                                         size_t *mac_length)
+{
+    hmac_sha256(key_buffer, key_buffer_size, input, input_length, mac);
+    *mac_length = 32;
+
+    (void)mac_size;
+    (void)attributes;
+    return PSA_SUCCESS;
```
[the remark this answers is not in the document]
Comment (Einhornhool) [reply sys/hashes/psa_riot_hashes/hmac_sha256.c:18]: As I wrote on the micro-ecc branch: Wouldn't it be better to check this on a higher level and make sure that the call never gets passed this far, if the input's not valid?

**Comment by Einhornhool** [comment]
To make this easier to test and work on are going to keep working on the original branch: https://github.com/RIOT-OS/RIOT/pull/18547

**Pull request closed without merging**